crates/uv-workspace/src/settings.rs

```diff
@@ -48,6 +47,39 @@ pub struct Options {
     pub override_dependencies: Option<Vec<pep508_rs::Requirement<VerbatimParsedUrl>>>,
 }
 
+/// Global settings, relevant to all invocations.
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct GlobalOptions {
+    pub native_tls: Option<bool>,
+    pub offline: Option<bool>,
+    pub no_cache: Option<bool>,
+    pub cache_dir: Option<PathBuf>,
+    pub preview: Option<bool>,
+}
+
+/// Shared settings, relevant to all dependency management operations.
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct InstallerOptions {
```

Comment:
I used the following terminology:

- "Compile" or "Compiler" for something that _just_ does resolution.
- "Sync" or "Syncer" for something that _just_ does installation (from an existing resolution).
- "Install" or "Installer" for something that does both resolution and installer.

In `cli.r`, the `CompilerArgs` are used for (e.g.) `uv pip compile` and `uv lock`. The `SyncerArgs` are used for (e.g.) `uv pip sync` and `uv sync`. The `InstallerArgs` are used for `uv pip install` and `uv run` and `uv tool run`.

Internally, the Project and Tool APIs just take `InstallerSettings`, so the commands get some settings they don't need. But it was a bit easier that way.

Reply:
I find it really weird that "Installer" isn't the thing that does _just_ installation. What about:

- "Resolver": _just_ does resolution
- "Installer": _just_ does installation
- "Complete": for both

Really not sure about the third name but the other two are way more obvious?

Reply:
I see the benefit of matching the command names, but I'm not sure it's quite worth it.

Reply:
I don't disagree but that's inconsistent with the `pip` API... The third name is what makes things really hard. I also considered `Operation` or something.

Reply:
I'd rather focus on internal consistency than consistency with the `pip` command names, I think.

Aren't you often taking the resolution and installation options and constructing the third "full" options from them anyway?

Reply:
I think that's going to disappear though based on https://github.com/astral-sh/uv/pull/4294#discussion_r1638293555.

Reply:
I'll do whatever. I can just run with "Complete" for now.

Reply:
I'm guessing there's a lot of overlap between installation and resolution and we shouldn't just do `Third(InstallationOptions, ResolutionOptions)`?

Reply:
If only there was an obvious third name.

Reply:
Yeah they mostly overlap but critically there are a few options that don't make sense for one vs. the other and I think exposing them on the CLI would be bad (like `uv lock` shouldn't have `--compile-bytecode`).

Reply:
Ok, I changed it to `Installer`, `Resolver`, and `Complete`.